### ISSUE TITLE
Implement small FC NN in cpp.

### ISSUE THREAD

**Issue by pyaf** [ninjakx/Coding-practice#6]
Implement a small neural network from scratch in cpp.
3 nodes => ReLu activation => 2 nodes => ReLU activation => 1 node => Sigmoid output
use mean square loss and gradient descent to optimize it. 

No need to do vectorization as of now.

**Comment by ninjakx** [comment]
https://github.com/ninjakx/Coding-practice/blob/master/NN.cpp
